x/subaccount/keeper/create_subaccount.go

```diff
@@ -0,0 +1,60 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/sge-network/sge/x/subaccount/types"
+)
+
+func (m msgServer) CreateSubAccount(
+	ctx context.Context,
+	request *types.MsgCreateSubAccount,
+) (*types.MsgCreateSubAccountResponse, error) {
+	sdkContext := sdk.UnwrapSDKContext(ctx)
+	moneyToSend := sdk.NewInt(0)
+
+	err := request.ValidateBasic()
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid request")
+	}
+
+	for _, balanceUnlock := range request.LockedBalances {
+		if balanceUnlock.UnlockTime.Unix() < sdkContext.BlockTime().Unix() {
+			return nil, types.ErrUnlockTokenTimeExpired
+		}
+
+		moneyToSend = moneyToSend.Add(balanceUnlock.Amount)
+	}
+
+	senderAccount, _ := sdk.AccAddressFromBech32(request.Sender)
+	subaccountOwner, _ := sdk.AccAddressFromBech32(request.SubAccountOwner)
+	if m.keeper.HasSubAccount(sdkContext, subaccountOwner) {
+		return nil, types.ErrSubaccountAlreadyExist
+	}
+
+	subaccountID := m.keeper.NextID(sdkContext)
+
+	// ALERT: If someone frontruns the account creation, will be overwritten here
+	subaccountAddress := types.NewAddressFromSubaccount(subaccountID)
+	address := m.accountKeeper.NewAccountWithAddress(sdkContext, subaccountAddress)
+	m.accountKeeper.SetAccount(sdkContext, address)
+
+	denom := m.keeper.GetParams(sdkContext).LockedBalanceDenom
+	err = m.bankKeeper.SendCoins(sdkContext, senderAccount, subaccountAddress, sdk.NewCoins(sdk.NewCoin(denom, moneyToSend)))
```

Comment:
Formerly, `OrderBook` module was responsible for transferring the tokens between accounts and module accounts using `funder` inderface. @3eyedraga Do you think that `subaccount` module should use them as well?

Reply:
As mentioned https://github.com/sge-network/sge/pull/208/files#r1259558460, this cannot be a module account, it's a normal base account.